pytorch_lightning/core/lightning.py

```diff
@@ -1948,3 +1949,30 @@ def model_size(self) -> float:
         size_mb = os.path.getsize(tmp_name) / 1e6
         os.remove(tmp_name)
         return size_mb
+
+    def add_to_queue(self, queue: torch.multiprocessing.SimpleQueue) -> None:
+        """Appends the :attr:`trainer.callback_metrics` dictionary to the given queue.
+
+        To avoid issues with memory sharing, we cast the data to numpy.
+
+        Args:
+            queue: the instance of the queue to append the data.
+        """
+        callback_metrics: dict = apply_to_collection(
+            self.trainer.callback_metrics, torch.Tensor, lambda x: x.cpu().numpy()
+        )  # send as numpy to avoid issues with memory sharing
+        queue.put(callback_metrics)
+
+    def get_from_queue(self, queue: torch.multiprocessing.SimpleQueue) -> None:
+        """Retrieve the :attr:`trainer.callback_metrics` dictionary from the given queue.
+
+        To preserve consistency, we cast back the data to ``torch.Tensor``.
+
+        Args:
+            queue: the instance of the queue from where to get the data.
+        """
+        # NOTE: `add_to_queue` needs to be called before
+        callback_metrics: dict = queue.get()
+        self.trainer.callback_metrics.update(
+            apply_to_collection(callback_metrics, np.ndarray, lambda x: torch.tensor(x))
+        )
```

Comment:
is this the only alternative to populate these metrics? why is this on a user the user interace of the lightning module? what happens if someone overrides this? is it meant to be overridden? 

it feels like the lightning module is used as a go-between between different parts of the trainer, in particular because the training type plugin technically has no reference to the trainer. 

structrually, we are repeatedly reaching through the lightning module to access the trainer in a very roundabout way. another example: https://github.com/PyTorchLightning/pytorch-lightning/blob/55a90af7fc0805855684e93dbad669f5bbe76eee/pytorch_lightning/plugins/training_type/sharded.py#L42-L57

it feels backwards and it also makes efforts like https://github.com/PyTorchLightning/pytorch-lightning/issues/7315 harder to work through when we keep relying on the trainer like this